clang/lib/Sema/SemaInit.cpp

```diff
@@ -4561,6 +4569,13 @@ static void TryConstructorInitialization(Sema &S,
 
   CXXConstructorDecl *CtorDecl = cast<CXXConstructorDecl>(Best->Function);
   if (Result != OR_Deleted) {
+    if (!IsListInit && Kind.getKind() == InitializationKind::IK_Default &&
```

Comment:
Ok, perhaps I'm missing something here... how do we know that a field marked with the attribute wasn't initialized here?  

Also, we have `DestRecordDecl`, so we have access to the `FieldDecl`s associated with it, right?

Reply:
>Because of the && DestRecordDecl->hasUninitializedExplicitInitFields() conditions a couple of lines below.

Right, but that is a property of the declaration 'elsewhere', not the initialization of the type here.  We actually shouldn't be storing that initialization on the  `RecordDecl` itself, that should be immutable once we see the semicolon.

>Trying to dig up the exact field is complicated and slow.
Complicated? Perhaps somewhat.  `slow` is usually OK for the purposes of diagnostics, it seems to be sensible here, IMO.

Reply:
This is more that this line of questioning is leading me to have some significant doubts as to whether this is doing what it is supposed to be doing, and doing so in an acceptable way.  

I'm not sure how the `RecordDecl` itself can contain the information as to whether this initialization is initializing everything necessary, and the goofy diagnostics here making the problem more opaque.  You don't have ot do them just yet (I want to discuss this with Aaron whether that field name needs to be identified), but unless I'm misunderstanding this (and I'll have to spend a few hours debugging to figure it out to convince me), I might be against this patch.

Reply:
You're just checking that it isnt list-init, but it could be copy/normal constructor init, right?  if you mean to catch 'default' initialization, don't you have to check the 'IK_Default' like on 4588?

Reply:
Ah,yes... no idea how I missed that... perhaps this is fine.  What about the case on 6477?